src/acq.c

```diff
@@ -96,8 +97,18 @@ enum bl_error bl_acq_start(
 		return BL_ERROR_ACTIVE_ACQUISITION;
 	}
 
-	/* For now, channels have a direct mapping to sources. */
-	uint16_t acq_chan_mask = src_mask;
+	bool is_flash_mode = (led_mask != 0x0000) &&
+	        ((led_mask & (led_mask - 1)) != 0);
+
+	if (is_flash_mode) {
+		enum bl_error status = bl_led_setup(led_mask);
+		if (status != BL_ERROR_NONE) {
+			return status;
+		}
+	}
+
+	/* Channels are combined by the LED channels and 3 non-LED sources */
+	uint32_t acq_chan_mask = led_mask | (src_mask & 0xF0) << 16;
```

Comment:
I think we should only do this in flash mode, not in continuous mode.

Reply:
why only in flash mode?

I think we still need to assign channel to sources, and enable channels correctly in continuous mode